docs/notary.md

```diff
@@ -0,0 +1,531 @@
+
+### Configuration
+
+To enable notary service node check firstly that
+[P2PSignatureExtensions](#NeoGo P2P signature extensions) are properly set up.
+Then add `P2PNotary` subsection to `ApplicationConfiguration` section of your
+node config.
+
+Parameters:
+* `Enabled`: boolean value, enables/disables the service node, `true` for service
+  node to be enabled
+* `UnlockWallet`: notary node wallet configuration:
+    - `Path`: path to NEP-6 wallet.
+    - `Password`: password for the account to be used by notary node.
+
+### Example
+
+```
+P2PNotary:
+  Enabled: true
+  UnlockWallet:
+    Path: "/notary_node_wallet.json"
+    Password: "pass"
+```
+
+
+## Notary request lifecycle guide
+
+Below presented stages each P2P signature collection request goes through. Use
+stages 1 and 2 to create, sign and submit P2P notary request. Stage 3 is
+performed by the notary service, do not require user's intervention and is given
+for informational purposes. Stage 4 contains tips to check notary request
+results.
+
+### 1. Notary deposit
+
+Notary service is not free and is paid through the sender's deposit that
+contained in the Notary native contract. Before the notary request will be
+submitted, you need to deposit enough GAS to the contract, otherwise, request
+won't pass verification.
+
+Notary native contract supports `onNEP17Payment` method, thus to deposit funds to
+the Notary native contract, transfer desired amount of GAS to the contract
+address. Use
+[func (*Client) TransferNEP17](https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.97.2/pkg/rpc/client#Client.TransferNEP17)
+with the `data` parameter matching the following requirements:
+- `data` should be an array of two elements: `to` and `till`.
+- `to` denotes the receiver of the deposit. It can be nil in case if `to` equals
+  to the GAS sender.
+- `till` denotes chain's height before which deposit is locked and can't be
+  withdrawn. `till` can't be set if you're not the deposit owner. Default `till`
+  value is current chain height + 5760. `till` can't be less than current chain
+  height. `till` can't be less than previously set value.
+
+Note, that first deposit call for the `to` address can't be less than 2×FEE GAS.
+Deposit is allowed for renewal, i.e. consequent `deposit` calls for the same `to`
+address add up specified amount to the already deposited value.
+
+After GAS transfer successfully submitted to the chain, use [Notary native
+contract API](#Native Notary contract) to manage your deposit.
+
+Note, that regular operation flow requires deposited amount of GAS to be
+sufficient to pay for the fallback transaction fees which are system fee and
+network fee. Fallback network fee includes (`NKeys`+1)×`FEE` = (0+1)×`FEE` = `FEE`
+GAS for `NotaryAssisted` attribute usaage and regular fee for the fallback size.
+If the deposited amount is not enough to pay the fallback fees, then
+`Insufficiend funds` error will be returned from the RPC node after notary
+request submission.
```

Comment:
Somehow note that this deposit limits all in-flight notary requests? The default deposit sum assumes one transaction in-flight, but if people need more they need to deposit more.